wagtail/admin/views/generic/history.py

```diff
@@ -60,8 +60,17 @@ class HistoryFilterSet(WagtailFilterSet):
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self.filters["action"].extra["choices"] = get_actions_for_filter(self.queryset)
-        self.filters["user"].extra["queryset"] = self.queryset.get_users()
+        actions = get_actions_for_filter(self.queryset)
+        if not actions:
+            del self.filters["action"]
+        else:
+            self.filters["action"].extra["choices"] = actions
+
+        users = self.queryset.get_users()
+        if not users.exists():
```

Comment:
I was hoping we could just evaluate the queryset here and use its length to determine this, and reuse the cached queryset to build the filter options. However, it seems django-filter (or maybe Django) clones the queryset so the cached results will not be reused and the query will be triggered again. We could hack it or work around it by turning it into a plain choice filter, but I think that's unnecessary complexity just to save a single `exists()` query.